src/packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -63,20 +63,32 @@ test('introspection --force', async () => {
 test('basic introspection with --url', async () => {
   ctx.fixture('introspection/sqlite')
   const introspect = new DbPull()
-  await introspect.parse(['--print', '--url', 'file:dev.db'])
+  const result = introspect.parse(['--print', '--url', 'file:dev.db'])
+  await expect(result).resolves.toBe('')
   expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchSnapshot()
 })
 
-test('basic introspection with invalid --url', async () => {
+test('basic introspection with invalid --url - empty host', async () => {
   ctx.fixture('introspection/sqlite')
   const introspect = new DbPull()
   const result = introspect.parse([
     '--print',
     '--url',
     'postgresql://root:prisma@/prisma',
   ])
+  await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+          Error parsing connection string: empty host in database URL
+
+        `)
+  expect(ctx.mocked['console.log'].mock.calls.join('\n')).toMatchSnapshot()
+})
+
+test('basic introspection with invalid --url', async () => {
+  ctx.fixture('introspection/sqlite')
+  const introspect = new DbPull()
+  const result = introspect.parse(['--print', '--url', 'invalidstring'])
   await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(
-    `Invalid data source URL, see https://www.prisma.io/docs/reference/database-reference/connection-urls`,
+    `Unknown database type invalidstring:`,
```

Comment:
What is the `:` for at the end of the message? Are there subsequent messages?

Reply:
Oh I see, it's interpreted a the protocol

Reply:
I add it because split removes it and it's needed to use the `protocolToDatabaseType` function